climada/engine/impact.py

```diff
@@ -1409,123 +1417,177 @@ def concat(cls, imp_list,concat_type='time_window'):
+                                 "The impacts are incompatible and cannot be concatenated.")
+            if len(set(units)) > 1:
+                raise ValueError(f"The given impacts have different units: {units}. "
+                                 "The impacts are incompatible and cannot be concatenated.")
+            unit = set(units[0])
+            return unit
+
+        def check_crs(imp_list):
+            crs = [imp.crs for imp in imp_list if imp.crs != '']
+            if len(crs) < len(imp_list):
+                raise ValueError(
+                    "Not all impacts have a CRS. Specify the same CRS for each impact."
+                    "The impacts are incompatible and cannot be concatenated.")
+            if len(set(crs)) > 1:
+                raise ValueError(f"The given impacts have different crs: {crs}. "
+                                 "The impacts are incompatible and cannot be concatenated.")
+            return crs[0]
+
+        def check_attributes(imp_list):
+            attributes = set.union(*[set(vars(imp).keys()) for imp in imp_list])
+            for attr_name in attributes:
+                if not all(hasattr(imp, attr_name) for imp in imp_list):
+                    raise ValueError(f"Attribute {attr_name} is not shared by all impacts. "
+                                    "The impacts are incompatible and cannot be concatenated.")
+            return True
+
+        def check_frequency(imp_list):
+            freqs = {imp.frequency[0] for imp in imp_list}
+            if len(freqs) > 1:
+                raise ValueError(f"Impacts do not have the same frequency: {freqs}."
+                                 "The impacts are incompatible and cannot be concatenated.")
+            freq = np.concatenate([imp.frequency for imp in imp_list], axis=0)
+            return freq
+
+        def check_dates(imp_list):
+
+            #compare dates of all impact pairs
+            for i, imp1 in enumerate(imp_list):
+                for imp2 in imp_list[i+1::]:
+                    dates_a_set = set(imp1.date)
+                    dates_b_set = set(imp2.date)
+                    if len(dates_a_set.intersection(dates_b_set)) > 0:
+                        raise ValueError("Found at least one duplicate date. "
+                                         "Impacts can not contain same dates."
+                                         "Impacts are incompatible and cannot be concatenated.")
+            date = np.array([date for imp in imp_list for date in imp.date])
+            return date
+
+        def check_exposure(imp_list):
+
+            #check total value
+            tot_vals = {imp.tot_value for imp in imp_list}
+            if len(tot_vals) > 1:
+                raise ValueError("The impacts are not based on same exposure."
+                                 "Total exposure values are different."
+                                 "The impacts are incompatible and cannot be concatenated.")
```

Comment:
I suggesst to make complete sentences as it reads much better.
```suggestion
                raise ValueError("The impacts are not based on the same exposure."
                                 "The total exposure values are different."
                                 "The impacts are incompatible and cannot be concatenated.")
```